### PR TITLE
#1778 Remove RegisterMessageSpecs

### PR DESCRIPTION
**Pull request by ccellado**
Currently message processing has a lot of flexibility we do not actually need for Ergo protocol client. Eliminate RegisterMessageSpecs and hard-code handlers in messageHandlers.

### CODE REVIEW

File: src/main/scala/org/ergoplatform/ErgoApp.scala [hunk 1]
```diff
@@ -96,14 +102,12 @@ class ErgoApp(args: Args) extends ScorexLogging {
       None
     }
 
-
   private val syncTracker = ErgoSyncTracker(scorexSettings.network, timeProvider)
-  private val statsCollectorRef: ActorRef = ErgoStatsCollectorRef(readersHolderRef, networkControllerRef, syncTracker, ergoSettings, timeProvider)
```
Comment (kushti) [inline src/main/scala/org/ergoplatform/ErgoApp.scala:101]: Like with ErgoNodeViewSynchronizer, we need to touch ErgoStatsCollectorRef to run it

File: src/main/scala/org/ergoplatform/ErgoApp.scala [hunk 2]
```diff
@@ -112,29 +116,69 @@ class ErgoApp(args: Args) extends ScorexLogging {
...
-  if (ergoSettings.scorexSettings.network.peerDiscovery) {
-    // Launching PeerSynchronizer actor which is then registering itself at network controller
-    PeerSynchronizerRef("PeerSynchronizer", networkControllerRef, peerManagerRef, scorexSettings.network)
-  }
+  private val messageHandlers: ActorRef => Map[MessageCode, ActorRef] =
+    networkControllerRef => {
+      val ergoNodeViewSynchronizerRef = ergoNodeViewSynchronizerRefPartial(
+        networkControllerRef
+      )
+      var map: Map[MessageCode, ActorRef] = Map(
```
Comment (kushti) [inline src/main/scala/org/ergoplatform/ErgoApp.scala:124]: GetPeers missed

File: src/main/scala/scorex/core/network/NetworkController.scala [hunk 3]
```diff
@@ -29,7 +29,8 @@ import scala.util.{Random, Try}
 class NetworkController(ergoSettings: ErgoSettings,
                         peerManagerRef: ActorRef,
                         scorexContext: ScorexContext,
-                        tcpManager: ActorRef
+                        tcpManager: ActorRef,
+                        messageHandlersPartial: ActorRef => Map[MessageCode, ActorRef]
```
Comment (kushti) [inline src/main/scala/scorex/core/network/NetworkController.scala:33]: There's no need in such flexibility probably, handlers can be just hardcoded in NetworkController

**Review by kushti — CHANGES_REQUESTED** [review]

File: src/main/scala/org/ergoplatform/ErgoApp.scala (hunk 1 above)
Comment (ccellado) [reply src/main/scala/org/ergoplatform/ErgoApp.scala:101]: This variable is untouched by PR

File: src/main/scala/scorex/core/network/NetworkController.scala (hunk 3 above)
Comment (ccellado) [reply src/main/scala/scorex/core/network/NetworkController.scala:33]: The problem is that code becomes very mess, if I try to create these actors inside NetworkController.
So in my opinion partial construction works better for readability.

**Review by kushti — APPROVED** [review]

**Pull request merged**